### ISSUE TITLE
[BUG] @beartype throws beartype.roar.BeartypeCallHintParamViolation even though manual isinstance works

### ISSUE THREAD

**Issue by f-fuchs** [beartype/beartype#411]
Hey,

I have a function that has the typehint `dict[str, torchmetrics.metric.Metric]` and I call it with an instance of `torchmetrics.classification.f_beta.BinaryF1Score`. When I do the `isinstance `check manually:

```
>>> import torchmetrics.classification
>>> from torchmetrics import Metric
>>> metric = torchmetrics.classification.BinaryF1Score()
>>> isinstance(metric,Metric)
True
```

I get a conformation that `BinaryF1Score` is an instance of `Metric` but my decorator throws following error:
`beartype.roar.BeartypeCallHintParamViolation: Function foundation_models.train.train() parameter metrics={'BinaryF1Score': <class 'torchmetrics.classification.f_beta.BinaryF1Score'>, 'Intersectio...'>} violates type hint dict[str, torchmetrics.metric.Metric], as dict key str 'BinaryF1Score' value <class "abc.ABCMeta"> <class 'torchmetrics.classification.f_beta.BinaryF1Score'> not instance of <protocol "torchmetrics.metric.Metric">.`

**Comment by leycec** [comment]
**Fascinating!** From the `beartype.roar.BeartypeCallHintParamViolation` exception message you've so kindly provided, it looks like your `foundation_models.train.train()` function is being passed a `metrics` dictionary with a `'BinaryF1Score'` key whose value is a **class** `torchmetrics.classification.f_beta.BinaryF1Score` rather than an **instance of that class.**

Is that the case? If so, that would *definitely* explain that exception. @beartype is probably right about a type-checking violation. A class isn't the same thing as an instance of that class, right? Well... not usually, anyway. Admittedly, Python can get pretty wacky. :laughing: 

If that isn't the case, would you mind providing a minimal working example (MWE) that reproduces this issue? I can't *quite* tell what exactly went wrong from the description you've so kindly provided. This is a "me" problem rather than a "you" problem. It's always me.

**Comment by f-fuchs** [comment]
hey, thanks for your fast response. you were indeed correct. I was not reading the error message correctly 😶

**Comment by leycec** [comment]
**Wonderful!** @beartype exception messages are probably a *lot* less readable than they should be. Confusion like this *kinda* is our fault. It is sad. :crying_cat_face:
